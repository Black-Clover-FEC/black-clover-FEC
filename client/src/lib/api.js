import axios from 'axios';
import config from '../env/config.js';


// This first section contains helper/template functions and constants to keep our other functions
// simple and DRY. Reuse them! But please don't edit them without team buy-in, because all
// of our individual API functions rely on them.
const api = {};

// URLs
const herokuUrl = `https://app-hrsei-api.herokuapp.com/api/fec2/${config.CAMPUS_CODE}`;
const unsplashOauthUrl = 'https://unsplash.com/oauth/';
const unsplashApiUrl = 'https://api.unsplash.com/';

// Headers
const herokuHeaders = {Authorization: config.AUTH};
const unsplashHeaders = {
  'Authorization': `Client-ID ${config.UNSPLASH_ACCESS_KEY}`,
  'Accept-Version': 'v1'
};
const unsplashTokenHeaders = {
  'Authorization': `Client-ID ${config.UNSPLASH_ACCESS_KEY}, Bearer ${config.UNSPLASH_USER_TOKEN}`,
  'Accept-Version': 'v1'
};
const imgurHeaders = {
  'Authorization': `Client-ID ${config.IMGUR_CLIENT_ID}`,
  'Content-Type': 'multipart/form-data'
};

// Atelier helper functions
const get = (endpoint, params = {}) => {
  return axios.get(`${herokuUrl}/${endpoint}/`, {headers: herokuHeaders, params: params})
    .then(results => results.data)
    .catch(err => console.error(`Error getting ${endpoint} from server:  ${err}`));
}

const post = (endpoint, data = {}, params = {}) => {
  return axios.post(`${herokuUrl}/${endpoint}/`, data, {headers: herokuHeaders, params: params})
    .then(results => results.data)
    .catch(err => console.error(`Error posting ${endpoint} to server:  ${err}`));
}

const put = (endpoint, data = {}, params = {}) => {
  return axios.put(`${herokuUrl}/${endpoint}/`, data, {headers: herokuHeaders, params: params})
    .then(results => results.data)
    .catch(err => console.error(`Error updating ${endpoint} on server:  ${err}`));
}



// FOR ALL BELOW FUNCTIONS, SEE ATELIER DOCS (ON LEARN) FOR LIST OF RELEVANT PARAMS TO PASS IN

// PRODUCTS
api.listProducts = (params) => {
  return get('products', params);
};

api.getProductId = (product_id) => {
  return get(`products/${product_id}`);
}

api.getProductStyles = (product_id) => {
  return get(`products/${product_id}/styles`);
}

api.getRelatedProducts = (product_id) => {
  return get(`products/${product_id}/related`);
}


// AGGREGATE DETAILS, STYLES, AND META DATA FOR PRODUCT

api.collectProductInfo = async (id) => {
  let product = {};
  product.details = await api.getProductId(id)
  product.styles = await api.getProductStyles(id);
  product.reviewsMeta = await api.getReviewsMetadata({product_id: id});
  return product;
}

// QUESTIONS AND ANSWERS
api.listQuestions = (params) => {
  return get('qa/questions', params);
}

api.listAnswers = (question_id, params) => {
  return get(`/qa/questions/${question_id}/answers`, params);
}

api.addQuestion = (params) => {
  return post('/qa/questions', params);
}

api.addAnswer = (question_id) => {
  return post(`/qa/questions/${question_id}/answers`, params);
}

api.markQuestionHelpful = (question_id) => {
  return put(`/qa/questions/${question_id}/helpful`);
}

api.reportQuesion = (question_id) => {
  return put(`/qa/questions/${question_id}/report`)
}

api.markAnswerHelpful = (answer_id) => {
  return put(`/qa/answers/${answer_id}/helpful`)
}

api.reportAnswer = (answer_id) => {
  return put(`/qa/answers/${answer_id}/report`)
}


// REVIEWS
api.listReviews = (params) => {
  return get('reviews', params)
    .then(data => data.results);
};

api.getReviewsMetadata = (params) => {
  return get('reviews/meta', params)
    .then(data => {
      let totalCount = 0;
      let totalScore = 0;
      for (let rating in data.ratings) {
        const count = parseInt(data.ratings[rating]);
        totalCount += count;
        totalScore += rating * count;
      }
      data.reviewsCount = totalCount;
      data.averageRating = totalScore / totalCount;
      data.recRate = data.recommended.true / totalCount;
      return data;
  })
  .catch(err => console.error(err));
};

api.addReview = (review) => {
  return post('reviews', review);
};

api.markReviewHelpful = (review_id) => {
  return put(`reviews/${review_id}/helpful`);
};

api.reportReview = (review_id) => {
  return put(`reviews/${review_id}/report`);
};


// CART
api.addCart = (sku_id) => {
  return put(`cart/${sku_id}`);

};

api.getCart = (params) => {
  return get('cart', params);
};


// INTERACTIONS
api.interact = (params) => {
  return post('iteraction', params);
};


// --------- UNSPLASH --------------

// oAuth

api.authorizeUser = () => {

  const params = {
    client_id: config.UNSPLASH_ACCESS_KEY,
    redirect_uri: 'urn:ietf:wg:oauth:2.0:oob',
    response_type: 'code',
    scope: 'public+read_photos+write_photos'
  };

  return axios.get(`${unsplashOauthUrl}authorize/`, {headers: unsplashHeaders, params: params})
    .then(res => console.log(res))
    .catch(err => console.error(err));
};

api.getToken = () => {

  const params = {
    client_id: config.UNSPLASH_ACCESS_KEY,
    client_secret: config.UNSPLASH_SECRET_KEY,
    redirect_uri: 'http://localhost:8080/',
    code: config.UNSPLASH_USER_AUTH_CODE,
    grant_type: 'authorization_code'
  };

  return axios.post(`${unsplashOauthUrl}token/`, params, {headers: unsplashHeaders})
    .then(res => console.log(res))
    .catch(err => console.error(err));
};

api.postPhoto = (photo) => {
  return axios.post(`https://api.imgur.com/3/image`, photo, {headers: imgurHeaders})
    .then(res => console.log(res))
    .catch(err => console.error(err));
};


export default api;