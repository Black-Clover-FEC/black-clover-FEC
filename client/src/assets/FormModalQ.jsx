import React, { useState } from 'react';
import StyleLib from './Stylesheet.jsx';
import styled from 'styled-components';
import Modal from 'react-modal';
import api from '../lib/api.js';

Modal.setAppElement('#root');

const FormModal = ({ product, isOpen, onClose, submitFunc }) => {

  const [photos, setPhotos] = useState([]);

  const handleFilesChange = (e) => {
    const file = e.target.files[0];
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = (e) => {
      let base64File = event.target.result;
      base64File = base64File.replace(/^data:image\/(png|jpg|jpeg|gif|heic|webp);base64,/, '');
      let photoForm = new FormData();
      photoForm.append('image', base64File);
      api.postPhotoToImgur(photoForm)
        .then(res => {setPhotos([...photos, res.data.link])})
        .catch(err => console.error(err));
    }
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    let formData = new FormData(e.target);
    formData = Object.fromEntries(formData.entries()); // converts from fromData object to JSON object.

    formData.product_id = product.id;
    console.log(formData);
    api.addQuestion(formData)
      .then(res => console.log('this is the response: ' + res)) // here for now for debugging purposes
      .then(() => {
        onClose();
        submitFunc();
      })
      .catch(err => console.error(err));
  }


  return (
    <Modal
      isOpen={isOpen}
      onRequestClose={onClose}
    >
      <StyleLib.h2>Ask your question</StyleLib.h2>
      <StyleLib.h5>about {product.name}</StyleLib.h5>
      <form onSubmit={handleSubmit} >

        <StyleLib.blockLabel>
          *Your Question <StyleLib.textarea name='body' rows='5' cols='30' maxlength='1000' required={true} />
        </StyleLib.blockLabel>

        <StyleLib.blockLabel>
          *Your nickname: <StyleLib.input name='name' maxlength='60' required={true} placeholder={"“Example: jackson11!”"}/>
          For privacy reasons, do not use your full name or email address
        </StyleLib.blockLabel>
        <StyleLib.blockLabel>
          *Your email: <StyleLib.input type='email' name='email' maxlength='60' required={true} />
          For authentication reasons, you will not be emailed
        </StyleLib.blockLabel>
        <StyleLib.button type='button' onClick={onClose}>Cancel</StyleLib.button>
        <StyleLib.button type='submit'>Submit</StyleLib.button>
      </form>
    </Modal>
  );
}

export default FormModal;