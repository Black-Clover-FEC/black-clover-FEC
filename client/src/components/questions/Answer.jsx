import React, {useState, useEffect} from 'react';
import api from '../../lib/api.js';
import StyleLib from '../../assets/Stylesheet.jsx';
import UserPhotos from '../../assets/UserPhotosQA.jsx';


const Answer = ({answer, helpfulCB}) => {
  // const photos = answer.photos;

  const [reportStatus, setReportStatus] = useState(false);

  const helpfulHandler = (e) => {
    api.markAnswerHelpful(answer.id)
      .then(res => helpfulCB())
      .catch(err => console.error(err));
  }

  const reportHandler = (e) => {
    api.reportAnswer(answer.id)
      .then(res => helpfulCB())
      .then(setReportStatus(true))
      .catch(err => console.error(err));
  }

  // const render = (photo, index) => {
  //   return <Photo photo={photo} key={index}/>
  // }
  return (
    <>
    {!reportStatus && answer && <section>
      <StyleLib.p>A: {answer.body}</StyleLib.p>
      <StyleLib.small>{answer.answerer_name} </StyleLib.small>
      <StyleLib.small>{(new Date(answer.date)).toDateString()} </StyleLib.small>
      <StyleLib.small>Helpful? </StyleLib.small>
      <StyleLib.linkButton onClick={helpfulHandler}>Yes </StyleLib.linkButton>
      <StyleLib.small>{answer.helpfulness} </StyleLib.small>
      <StyleLib.linkButton onClick={reportHandler}>Report</StyleLib.linkButton>
      {answer.photos ? <UserPhotos photos={answer.photos}/> : null}
    </section>}
    </>
  )
}

export default Answer;