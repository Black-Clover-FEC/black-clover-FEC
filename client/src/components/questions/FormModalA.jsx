import React, { useState } from 'react';
import StyleLib from '../../assets/Stylesheet.jsx';
import styled from 'styled-components';
import Modal from 'react-modal';
import api from '../../lib/api.js';

Modal.setAppElement('#root');

const FormModal = ({ product, question, answers, isOpen, onClose, setAnswers }) => {

  const [photos, setPhotos] = useState([]);

  // const handleFilesChange = (e) => {
  //   const file = e.target.files[0];
  //   const reader = new FileReader();
  //   reader.readAsDataURL(file);
  //   reader.onload = (e) => {
  //     let base64File = event.target.result;
  //     base64File = base64File.replace(/^data:image\/(png|jpg|jpeg|gif|heic|webp);base64,/, '');
  //     let photoForm = new FormData();
  //     photoForm.append('image', base64File);
  //     api.postPhotoToImgur(photoForm)
  //       .then(res => {setPhotos([...photos, res.data.link])})
  //       .catch(err => console.error(err));
  //   }
  // }


  const handleSubmit = (e) => {
    e.preventDefault();
    let formData = new FormData(e.target);
    formData = Object.fromEntries(formData.entries()); // converts from fromData object to JSON object.

    api.addAnswer(question.question_id, formData)
      .then(res => console.log('this is the response: ' + res)) // here for now for debugging purposes
      .then(() => {
        api.listAnswers(question.question_id, {
          page: 1,
          count: 1000
        })
          .then(data => {
            setAnswers(data.results)
            console.log('data.results', data.results)
            console.log('data', data)
            console.log('answers', answers)
          })
        .catch(err => console.error(err));
        onClose()
        console.log(answers)
      })
      .catch(err => console.error(err));
  }


  return (
    <Modal
      isOpen={isOpen}
      onRequestClose={onClose}
    >
      <StyleLib.h2>Submit your answer</StyleLib.h2>
      <StyleLib.h5>{product.name}: {question.question_body}</StyleLib.h5>
      <form onSubmit={handleSubmit} >

        <StyleLib.blockLabel>
          *Your Answer <StyleLib.textarea name='body' rows='5' cols='30' maxlength='1000' required={true} />
        </StyleLib.blockLabel>

        {/* {photos.length < 5 && (
          <StyleLib.blockLabel>
            Upload photos:
            <StyleLib.input name='photos' type='file' accept='image/*, .heic' onChange={handleFilesChange} multiple={true} />
          </StyleLib.blockLabel>
        )} */}

        <StyleLib.blockLabel>
          *Your nickname: <StyleLib.input name='name' maxlength='60' required={true} placeholder={"“Example: jackson11!”"}/>
          For privacy reasons, do not use your full name or email address
        </StyleLib.blockLabel>
        <StyleLib.blockLabel>
          *Your email: <StyleLib.input type='email' name='email' maxlength='60' required={true} />
          For authentication reasons, you will not be emailed
        </StyleLib.blockLabel>
        <StyleLib.button type='button' onClick={onClose}>Cancel</StyleLib.button>
        <StyleLib.button type='submit'>Submit</StyleLib.button>
      </form>
    </Modal>
  );
}

export default FormModal;