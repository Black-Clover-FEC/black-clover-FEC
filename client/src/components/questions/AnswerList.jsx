import React, {useState, useEffect} from 'react';
import Answer from './Answer.jsx';
import api from '../../lib/api.js';
import StyleLib from '../../assets/Stylesheet.jsx';

const AnswerList = ({answers, result, product, question, helpfulCB}) => {
  const answersIDArray = Object.keys(answers);
  const answersArray = [];
  for (var i = 0; i < answersIDArray.length; i++ ) {
    answersArray.push(answers[answersIDArray[i]]);
  }
  const [displayedAnswers, setDisplayedAnswers] = useState([]);
  const [sortedAnswersArray, setSortedAnswersArray] = useState(answersArray.sort((a, b) => b.helpfulness - a.helpfulness));
  const [numberOfAnswers, setNumberOfAnswers] = useState(2);
  const [buttonText, setButtonText] = useState('');

  useEffect(() => {setDisplayedAnswers(sortedAnswersArray.slice(0, 2))}, []);
  useEffect(() => {sortAnswers()}, []);
  useEffect(() => {sortAnswers()}, [numberOfAnswers])
  useEffect(() => {
    updateDisplayedAnswers();
  }, [numberOfAnswers]);

  useEffect(() => {
    buttonTextHelper();
  }, [displayedAnswers]);

  const render = (answer, index) => {
    return <Answer answer={answer} answersArray={answersArray} sortAnswers={sortAnswers} result={result} product={product} question={question} helpfulCB={helpfulCB} key={index}/>
  }

  const sortAnswers = () => {
    setSortedAnswersArray(answersArray.sort((a, b) => b.helpfulness - a.helpfulness));
  }

  const updateDisplayedAnswers = () => {
    setDisplayedAnswers(sortedAnswersArray.slice(0, numberOfAnswers));
  }

  const updateNumberOfAnswers = (e) => {
    if (numberOfAnswers < answersArray.length) {
      setNumberOfAnswers(answersArray.length);
    } else {
      setNumberOfAnswers(2);
    };
  };

  const buttonTextHelper = () => {
    if (displayedAnswers.length < answersArray.length) {
      setButtonText('See More Answers');
    }
    if (displayedAnswers.length >= answersArray.length) {
      setButtonText('Collapse Answers');
    }
  }

  return (
    <section>
      {displayedAnswers.map((answer, index) => render(answer, index))}
      {answersArray.length > 2 && <StyleLib.button onClick={updateNumberOfAnswers}>{buttonText}</StyleLib.button>}
    </section>
  )
}

export default AnswerList;