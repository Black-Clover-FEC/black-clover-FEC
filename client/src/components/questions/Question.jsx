import React, {useEffect, useState} from 'react';
import AnswerList from './AnswerList.jsx';
import api from '../../lib/api.js';
import StyleLib from '../../assets/Stylesheet.jsx';

const Question = ({result, helpfulCB}) => {
  const answers = result.answers;
  const [helpfulQCount, setHelpfulQCount] = useState(result.question_helpfulness);

  const helpfulHandler = (e) => {
    api.markQuestionHelpful(result.question_id)
      .then(res => helpfulCB())
      .then(setHelpfulQCount(result.question_helpfulness))
      .catch(err => console.error(err));
  }

  const reportHandler = (e) => {
    api.reportQuestion(result.question_id)
      .then(res => helpfulCB())
      .catch(err => console.error(err));
  }

  return (
    <StyleLib.tile>
      <StyleLib.p>Q: {result.question_body}</StyleLib.p>
      <StyleLib.small>{(new Date(result.question_date)).toDateString()} </StyleLib.small>
      <StyleLib.small>Helpful?
        <StyleLib.linkButton onClick={helpfulHandler}>
          Yes
        </StyleLib.linkButton>
        {helpfulQCount}
      </StyleLib.small>
      <StyleLib.linkButton onClick={reportHandler}>Report </StyleLib.linkButton>

      <AnswerList answers={answers} helpfulCB={helpfulCB}/>
    </StyleLib.tile>
  )
}

export default Question;