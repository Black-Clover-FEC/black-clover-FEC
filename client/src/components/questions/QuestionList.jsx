import React, {useState, useEffect} from 'react';
import Question from './Question.jsx';
import api from '../../lib/api.js';
import StyleLib from '../../assets/Stylesheet.jsx';

const QuestionList = ({displayedResults}) => {
  const render = (result, index) => {
    return <Question result={result} key={index}/>
  }
  return (
    <section>
      {displayedResults.map((result, index) => render(result, index))}
    </section>
  )
}

export default QuestionList;