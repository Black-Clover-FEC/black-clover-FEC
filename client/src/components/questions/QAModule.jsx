import React, {useState, useEffect} from 'react';
import StyleLib from '../../assets/Stylesheet.jsx';
import api from '../../lib/api.js';
import ReviewAndQuestionList from '../../assets/ReviewAndQuestionList.jsx';
import QuestionList from './QuestionList.jsx';
import FormModalQ from '../../assets/FormModalQ.jsx'

const QAModule = ({product}) => {
  const productID = product.id;

  // React Hooks
  const [numberOfQuestions, setNumberOfQuestions] = useState(2);
  const [displayedResults, setDisplayedResults] = useState([]);
  const [productResults, setProductResults] = useState([]);
  const [modal, setModal] = useState(false);
  const [input, setInput] = useState('');
  const [search, setSearch] = useState([]);

  // On Render Effects
  useEffect(() => {getQuestions()}, []);
  useEffect(() => {updateDisplayedResults()}, []);
  useEffect(() => {updateDisplayedResults(), sortResults()}, [numberOfQuestions, search]);
  useEffect(() => {filterResults()}, [input]);

  // Helper Functions
  const getQuestions = () => {
    api.listQuestions({
      product_id: productID,
      page: 1,
      count: 900
    })
      .then(data => {
        setProductResults(data.results)
        setSearch(data.results.filter(product => product.question_body.includes(input)))
        setDisplayedResults(data.results.slice(0, numberOfQuestions))
      })
      .catch(err => console.error(err))
  }

  const sortResults = () => {
    setSearch(search.sort((a, b) => b.helpfulness - a.helpfulness))
  }

  const inputTracker = (e) => {
    if (e.target.value.length >= 3) {
      setInput(e.target.value);
    } else {
      setInput('');
    }
  }

  const filterResults = () => {
    setSearch(productResults.filter(product => product.question_body.includes(input)));
  }

  const showMoreHandler = (e) => setNumberOfQuestions(numberOfQuestions + 2);
  const updateDisplayedResults = () => setDisplayedResults(search ? search.slice(0, numberOfQuestions) : productResults.slice(0, numberOfQuestions));
  const openModal = () => setModal(true);
  const closeModal = () => setModal(false);

  // Rendered Elements
  return (
    <section>
      <StyleLib.tile>
        <StyleLib.h2>Questions and Answers</StyleLib.h2>
        <StyleLib.searchBar placeholder={"Have a question? Search for answersâ€¦"} onChange={(e) => inputTracker(e)}/>
        <QuestionList displayedResults={displayedResults} helpfulCB={getQuestions} product={product}/>
        {(search ? (displayedResults.length < search.length) : (displayedResults.lenght < productResults.length)) && <StyleLib.button onClick={showMoreHandler}>More Answered Questions</StyleLib.button> }
        <StyleLib.button onClick={openModal}>Add A Question</StyleLib.button>
      </StyleLib.tile>

      <FormModalQ
        product={product}
        isOpen={modal}
        onClose={closeModal}
        submitFunc={getQuestions}
         />
    </section>
  )
}

export default QAModule;