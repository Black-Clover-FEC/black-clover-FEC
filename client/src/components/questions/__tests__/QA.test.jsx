import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';  // used to assert and manipulate rendered components
import ProdMock from '../../../../../__mocks__/productMock.js';
import MockFunc from '../../../../../__mocks__/functionMocks.js';
import QuestionMock from '../../../../../__mocks__/questionMock.js';

import Answer from '../Answer.jsx';
import AnswerList from '../AnswerList.jsx';
import QAModule from '../QAModule.jsx';
import Question from '../Question.jsx';
import QuestionList from '../QuestionList.jsx';
import FormModalQ from '../../../assets/FormModalQ.jsx';
import FormModalA from '../../../assets/FormModalA.jsx';
import StyleLib from '../../../assets/Stylesheet.jsx';
import api from '../../../lib/api.js';

describe('Does a component render to the page', () => {

  it('tests to see if QuestionList component renders correctly', () => {
    const tree = renderer.create(<QuestionList displayedResults={QuestionMock.results} helpfulCB={MockFunc} product={ProdMock}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('tests to see if Question component renders correctly', () => {
    const tree = renderer.create(<Question result={QuestionMock.results[0]} product={ProdMock} helpfulCB={MockFunc} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('tests to see if AnswerList component renders correctly', () => {
    const tree = renderer.create(<AnswerList answers={QuestionMock.results[0].answers} helpfulCB={MockFunc} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('tests to see if Answer component renders correctly', () => {
    const tree = renderer.create(<Answer answer={QuestionMock.results[0].answers[0]} helpfulCB={MockFunc} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('tests to see if FormModalQ component renders correctly', () => {
    const tree = renderer.create(<FormModalQ product={ProdMock} isOpen={true} onClose={MockFunc.default} submitFunc={MockFunc.default} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('tests to see if FormModalA component renders correctly', () => {
    const tree = renderer.create(<FormModalA
       product={ProdMock}
       question={QuestionMock.results[0]}
       isOpen={true}
       onClose={MockFunc.default}
       submitFunc={MockFunc.default}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('tests to see if StyleLib.searchBar component renders correctly', ()=> {
    const tree = renderer.create(<StyleLib.searchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('test helpfulHandler', () => {
    const tree = renderer.create(<Answer answer={QuestionMock.results[0].answers[5991165]} helpfulCB={MockFunc.default} />).toJSON();
    tree.getInstance().helpfulHandler();
  })



})