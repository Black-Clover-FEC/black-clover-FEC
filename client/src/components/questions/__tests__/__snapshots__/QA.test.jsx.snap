// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does a component render to the page tests to see if Answer component renders correctly 1`] = `null`;

exports[`Does a component render to the page tests to see if AnswerList component renders correctly 1`] = `
<section>
  <button
    className="sc-fsQiph beTiCm"
    onClick={[Function]}
  />
</section>
`;

exports[`Does a component render to the page tests to see if FormModalA component renders correctly 1`] = `null`;

exports[`Does a component render to the page tests to see if FormModalQ component renders correctly 1`] = `null`;

exports[`Does a component render to the page tests to see if Question component renders correctly 1`] = `
<li
  className="sc-fbJfA fZxUVg"
>
  <p
    className="sc-gLDzan ZmgyA"
  >
    Q: 
    Is this LEGIT?
  </p>
  <small
    className="sc-iAEyYk kbAVNF"
  >
    Thu Mar 23 2023
     
  </small>
  <small
    className="sc-iAEyYk kbAVNF"
  >
    Helpful?
    <button
      className="sc-qRumB krmTCT"
      onClick={[Function]}
    >
      Yes
    </button>
    4
  </small>
  <button
    className="sc-qRumB krmTCT"
    onClick={[Function]}
  >
    Report 
  </button>
  <button
    className="sc-qRumB krmTCT"
    onClick={[Function]}
  >
    Add Answer
  </button>
  <section>
    <section>
      <p
        className="sc-gLDzan ZmgyA"
      >
        A: 
        a
      </p>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        nickynickyjam
         
      </small>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        Thu Mar 23 2023
         
      </small>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        Helpful? 
      </small>
      <button
        className="sc-qRumB krmTCT"
        data-testid="helpfulHandler"
        onClick={[Function]}
      >
        Yes 
      </button>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        6
         
      </small>
      <button
        className="sc-qRumB krmTCT"
        data-testid="reportHandler"
        onClick={[Function]}
      >
        Report
      </button>
      <ul>
        <img
          className="sc-ktEKTO cwglxa"
          onClick={[Function]}
          src="https://orioledogclub.files.wordpress.com/2014/07/construction-dog.jpg"
        />
      </ul>
    </section>
    <section>
      <p
        className="sc-gLDzan ZmgyA"
      >
        A: 
        Of course!
      </p>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        Seller
         
      </small>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        Fri Mar 24 2023
         
      </small>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        Helpful? 
      </small>
      <button
        className="sc-qRumB krmTCT"
        data-testid="helpfulHandler"
        onClick={[Function]}
      >
        Yes 
      </button>
      <small
        className="sc-iAEyYk kbAVNF"
      >
        9
         
      </small>
      <button
        className="sc-qRumB krmTCT"
        data-testid="reportHandler"
        onClick={[Function]}
      >
        Report
      </button>
      <ul />
    </section>
    <button
      className="sc-fsQiph beTiCm"
      onClick={[Function]}
    >
      See More Answers
    </button>
  </section>
</li>
`;

exports[`Does a component render to the page tests to see if QuestionList component renders correctly 1`] = `
<section>
  <li
    className="sc-fbJfA fZxUVg"
  >
    <p
      className="sc-gLDzan ZmgyA"
    >
      Q: 
      Is this LEGIT?
    </p>
    <small
      className="sc-iAEyYk kbAVNF"
    >
      Thu Mar 23 2023
       
    </small>
    <small
      className="sc-iAEyYk kbAVNF"
    >
      Helpful?
      <button
        className="sc-qRumB krmTCT"
        onClick={[Function]}
      >
        Yes
      </button>
      4
    </small>
    <button
      className="sc-qRumB krmTCT"
      onClick={[Function]}
    >
      Report 
    </button>
    <button
      className="sc-qRumB krmTCT"
      onClick={[Function]}
    >
      Add Answer
    </button>
    <section>
      <button
        className="sc-fsQiph beTiCm"
        onClick={[Function]}
      />
    </section>
  </li>
</section>
`;

exports[`Does a component render to the page tests to see if StyleLib.searchBar component renders correctly 1`] = `
<input
  className="sc-kFuwaP ekhEca"
/>
`;
